src/Controller/Api/NodeGeneralInfoController.ts

```diff
@@ -0,0 +1,46 @@
+import {Request, Response} from "express";
+import {ValidatedRequest} from "express-joi-validation";
+
+import {NodeGeneralInfoService} from "../../Services/NodeGeneralInfoService";
+import {CreateNodeGeneralInfoRequestSchema} from "./NodeGeneralInfoControllerValidation";
+import logger from "../../Services/Logger";
+
+export class NodeGeneralInfoController {
+
+    private nodeGeneralInfoService: NodeGeneralInfoService;
+
+    constructor(nodeGeneralInfoService: NodeGeneralInfoService) {
+        this.nodeGeneralInfoService = nodeGeneralInfoService;
+    }
+
+    public async updateOrCreateNodeGenralInfo(
+        req: ValidatedRequest<CreateNodeGeneralInfoRequestSchema>,
+        res: Response) {
+        try {
+            const {version, sectorSize, minerPower, totalPower} = req.body;
+            const nodeId = res.locals.node.id;
+            const result = await this.nodeGeneralInfoService.updateOrCreateNodeGenralInfo(
+                version, sectorSize, minerPower, totalPower, nodeId);
+            if (result) {
```

Comment:
Can this be null? If yes, this request will end up hanging.
If not, you can write without if and just return result.